### PR TITLE
:sparkles: Use kairos-sdk module

### PR DESCRIPTION
**Pull request by Itxaka**
**What this PR does / why we need it**:
Exports the sdk into its own module so it can be released, fixed and worked without the kairos dependency and instead have kairos depend on it

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by Itxaka** [comment]
IMHO the machine module under pkg should also be moved into the sdk. Plus it seems that there are some duplicated functions like BootFrom which checks where it has booted from and the runtime also provides the same.

File: pkg/config/config.go [hunk 1]
```diff
@@ -12,10 +12,10 @@ import (
 	retry "github.com/avast/retry-go"
 	"github.com/imdario/mergo"
 	"github.com/itchyny/gojq"
+	"github.com/kairos-io/kairos-sdk/bundles"
+	"github.com/kairos-io/kairos-sdk/unstructured"
 	schema "github.com/kairos-io/kairos/pkg/config/schemas"
```
Comment (mudler) [inline pkg/config/config.go:17]: wondering, shall we make the config part of the sdk? in any case doesn't matter with this PR

**Review by mudler — DISMISSED** [review]
Looking good!

File: cmd/agent/main.go [hunk 2]
```diff
@@ -9,12 +9,12 @@ import (
 	"github.com/kairos-io/kairos/internal/bus"
 	"github.com/kairos-io/kairos/internal/webui"
 
+	bundles "github.com/kairos-io/kairos-sdk/bundles"
```
Comment (mudler) [inline cmd/agent/main.go:12]: maybe we should have named the repo just "sdk" and make the golang files top level? so imports are cleaner:
```
import (
 "github.com/kairos-io/sdk/..."
)
```

**Comment by mudler** [comment]
> IMHO the machine module under pkg should also be moved into the sdk. Plus it seems that there are some duplicated functions like BootFrom which checks where it has booted from and the runtime also provides the same.

yep, definetly :+1: feel free to take a stab at it

**Review by mudler — APPROVED** [review]

File: pkg/config/config.go (hunk 1 above)
Comment (Itxaka) [reply pkg/config/config.go:17]: yes, indeed, it would be the perfect part.

File: cmd/agent/main.go (hunk 2 above)
Comment (Itxaka) [reply cmd/agent/main.go:12]: Thougth about that but feel like it was very...generic. Like if we had down the line a different sdk for something else? kairos-sdk looked like it was very clear what it is for. I dunno...we have time to change this, I have no strong feelings one way or the other

**Pull request merged**